resin_release_tool/releaser.py

```diff
@@ -1,19 +1,64 @@
 from functools import lru_cache
-from balena import Balena
+import click
+from balena import Balena, Settings
+from resin_release_tool.balena_backend import BalenaBackend
 from collections import defaultdict
 
 
 class BalenaReleaser:
-    def __init__(self, token, app_id):
+    def __init__(self, token, app_id, balena_backend=None):
         self.balena = Balena()
         self.balena.auth.login_with_token(token)
 
         self.models = self.balena.models
+        if balena_backend is None:
+            balena_backend = BalenaBackend(
+                models=self.models, settings=self.balena.settings
+            )
+        self.balena_backend = balena_backend
+        self._check_sdk_version()
 
         self.app_id = app_id
 
+    def _check_sdk_version(self):
+        """The Balena sdk creates on installation a config file which sets the version to use for this user"""
+        try:
+            configured_version = self.balena.settings.get("api_version")
+            default_version = Settings._setting.get("api_version")
+            if (
+                not configured_version == BalenaBackend.SUPPORTED_API_VERSION
+                or default_version != BalenaBackend.SUPPORTED_API_VERSION
+            ):
```

Comment:
This `not configured_version == ...` is acting as `configured_version != ...`, right? The latter should be easier to follow the logic of (to me at least).

Reply:
you are right, I changed it to != for both cases